### DISCUSSION TITLE
General Feedback

### DISCUSSION THREAD

**Discussion by 3Diva** [MoonRide303/Fooocus-MRE#43] [General]
I wanted to leave some feedback on this version. After working with this version of Fooocus for a few hours I have to say that I'm really enjoying it! It has several features that I was missing and I appreciate being able to use them with the sleek stream-lined UI of Fooocus! 

I was a little confused at first about how to work the IMG2IMG functions, having both Img2Img and Revision in the same tab is a bit confusing. Ideally, I think a seperate tab for Revision would be awesome and a lot less confusing.

I LOVE that you added ControlNet Canny! Wow! SO HELPFUL! Thank you!

**A couple of QOL updates that I think would be fantastic:** 

- Is there a way to have it remember our settings? Every time I load it up I have to change the Base Model, Refiner, Lora, Resolution, Image Number, etc before getting started. Is there a way to save those settings? I'm thinking probably not, but thought I'd ask just in case! 😄 (EDIT: It does now by modifying the "settings-example" JSON and renaming it as "settings"! AWESOME! Thank you for the great update, MoonRide303!)
- I would LOVE it if there was an after generation face and/or hands fix like ADetailer. SDXL is awesome but it still struggles with hands and faces sometimes, particularly when not generating closeup images. ADetailer has been a massive help with this in A1111 and having that functionality in Fooocus would be amazing!

So far I'm REALLY loving Fooocus vanilla as well as Fooocus-MRE. I appreciate all the hard work that you and lllyasviel have put into Fooocus and the different versions! Thank you!

**Comment by barepixels** [comment]
About remember settings

1) duplicate "settings-example.json" and rename the duplicate file as "settings.json"
2) edit "settings.json"

vote for ADetailer

**Reply by 3Diva** [reply]
@barepixels  Yeah the inclusion of the "settings" option is AWESOME! I'm so happy that got added! It's been a big help. Also the "paths" option has been incredibly helpful as well. I no longer need separate folders with checkpoints and loras for the Fooocus-MRE, Automatic1111, and ComfyUI - I can have one folder for all the checkpoints and use the "paths" file to point to that folder. That's a HUGE hard drive space saver!

**Comment by osquitarxp** [comment]
Please help, I was working with this great version that seemed fantastic and intuitive to use. Now I get an error when starting:

missing {'cond_stage_model.clip_g.transformer.text_model.embeddings.position_ids'}
Refiner model loaded: C:\Difusion\Focus Mre\Fooocus-MRE\models\checkpoints\sd_xl_refiner_1.0_0.9vae.safetensors
Exception in thread Thread-2 (worker):
Traceback (most recent call last):
  File "C:\Difusion\Focus Mre\Fooocus-MRE-env\lib\threading.py", line 1016, in _bootstrap_inner
    self.run()
  File "C:\Difusion\Focus Mre\Fooocus-MRE-env\lib\threading.py", line 953, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Difusion\Focus Mre\Fooocus-MRE\modules\async_worker.py", line 40, in worker
    import modules.default_pipeline as pipeline
  File "C:\Difusion\Focus Mre\Fooocus-MRE\modules\default_pipeline.py", line 269, in <module>
    refresh_everything(
  File "C:\Difusion\Focus Mre\Fooocus-MRE\modules\default_pipeline.py", line 258, in refresh_everything
    virtual_memory.try_move_to_virtual_memory(xl_refiner.unet.model)
  File "C:\Difusion\Focus Mre\Fooocus-MRE-env\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Difusion\Focus Mre\Fooocus-MRE\modules\virtual_memory.py", line 185, in try_move_to_virtual_memory
    move_to_virtual_memory(model, comfy_unload)
  File "C:\Difusion\Focus Mre\Fooocus-MRE-env\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "C:\Difusion\Focus Mre\Fooocus-MRE\modules\virtual_memory.py", line 75, in move_to_virtual_memory
    model_management.unload_model()
AttributeError: module 'comfy.model_management' has no attribute 'unload_model'

I must say that I also have the original version of fooocus installed and it works ok without errors, I also have the stable diffusion 1111 and it is ok, but this version of foocus MRE worked fine for a few days, and then it failed... I deleted everything and returned it to install from scratch step by step.... and if sending the same error.... ! someone help me please!

**Reply by 3Diva** [reply]
I wish I could help, but all that is very much above my head. :( Did you ever get it figured out?

**Reply by MoonRide303** [reply]
@3Diva It's fixed in v2.0.19 release. This was problem for happening about 2 days, when I pulled changes from vanilla that included Virtual Memory system that wasn't compatible with latest Comfy used in MRE. And this VM thing activated itself only on machines having less than 32 GB RAM, so sadly I didn't notice it on my main PC I use for development.

**Comment by 3Diva** [comment]
I don't know how you do it @MoonRide303 but somehow your Fooocus-MRE generates images faster than ComfyUI, Automatic1111, AND the original Fooocus! In particular the first time you generate an image. It takes MUCH longer for the other programs to generate the first image, usually taking SEVERAL MINUTES for the model/checkpoint to load and for the image to start generating. But I've noticed that with Fooocus-MRE it takes a MUCH sorter time for the first image to start to generate. I don't know how you do it - but THANK YOU! That's a major pain with the others taking so long to generate the first image. I find myself using Fooocus-MRE more and more because of the speed in which it produces images as well as how good the images look! Thank you, sir! Your work is AMAZING!

**Reply by barepixels** [reply]
Should thanks lllyasviel too lolol

**Reply by 3Diva** [reply]
@barepixels I do, frequently, but in this case I'm particularly pointing out something that Fooocus-MRE seams to do better. I'm sure Illyasviel appreciates your white knighting though. :P 😜 lol

**Reply by MoonRide303** [reply]
@3Diva Glad to hear it's working well, but @barepixels is right - it's combination of optimizations done in both Comfy (MRE uses newer version than one included in vanilla), and what lllyasviel did in the vanilla. I just put it together, and disabled stuff that weren't working (like VM from vanilla - which is not really needed with new Comfy backend).

And I noticed recently that lllyasviel also started adding some QoL changes that were available in MRE into vanilla - like stop button, or paths to model files, so vanilla should be in a pretty good shape soon, too :).

I am just not sure about those UI changes since vanilla v2.0.50 - it just looks bad to me with parts of UI being rendered out of screen space, and I don't want MRE to look like that. Inpainting, outpainting and upscaling quality in current vanilla is also pretty low, and I don't want to include features that produce low quality output.

**Reply by lanyusan** [reply]
@MoonRide303

I suggest to develop an experimental branch that is in sync with lllyasiel's  progress for users who is ok with his UI and current quality of generated results.

Otherwise your fans, me included might be wondering if you will keep up with his releases.

**Reply by barepixels** [reply]
While I have vanilla installed to test.  It will never be my work horse because the lack of embedded metadata

**Reply by 3Diva** [reply]
@lanyusan I think that @MoonRide303 is right to wait a bit and to not try and stay in lock-step with vanilla Fooocus, as it takes time to work out some of the bugs and quality issues that can come with implementing new features. Giving lllyasviel time to work out the bugs with vanilla Fooocus means less changes that MoonRide303 will need to make. I honestly think that the pace of production and updates with Fooocus-MRE has been amazing!

**Reply by lanyusan** [reply]
Yes I agree with you and @MoonRide303 to wait out for a while.

My concern is that a lag too long would make some people wonder if this fork is till being actively maintained. I mean this fork has  got some momentum and a fan base. It is important to keep it going.

**Reply by MoonRide303** [reply]
@lanyusan I've merge latest vanilla changes (up to v2.0.76) on the [testing](https://github.com/MoonRide303/Fooocus-MRE/tree/testing) branch of MRE:
![image](https://github.com/MoonRide303/Fooocus-MRE/assets/130458190/09c6c61a-8278-428e-883a-ef264426833b)

and basic use cases seem to work - but I want to do some more testing before pushing it to users as update.

**Comment by lanyusan** [comment]
Great. I have already been using your testing branch. 

The original Foocus crashes often on my Ubuntu PC due to the virtual memory problem.

Yours doesn't have this problem.

**Comment by Axxel425** [comment]
Hi @MoonRide303,  congrats for your great fooocus MRE, I enjoy to use it since the start, btw I have a little problem with your last release , each time at start the file "user_path_config.txt"  returns to default settings , so I can't use my own paths. thank you for your help.
(paths.json is ok but doesn't seem to have any effects with the last release )

**Reply by MoonRide303** [reply]
@Axxel425 My bad, not enough testing after merging changes from vanilla - fixed in v2.0.78.1.

PS In case of any update problems you can just use last release zip (v2.0.19), and start it with update from fresh folder - it helped in few cases.

**Reply by Axxel425** [reply]
ty ;)
